packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -48,6 +48,48 @@ export function isTreeNodeSchemaClass<
  */
 export type AllowedTypes = readonly LazyItem<TreeNodeSchema>[];
 
+/**
+ * Additional information to provide to a Node Schema.
+ *
+ * @typeParam TCustomMetadata - Custom metadata properties to associate with the Node Schema.
+ * See {@link NodeSchemaMetadata.custom}.
+ *
+ * @public
+ */
+export interface NodeSchemaProps<out TCustomMetadata = unknown> {
```

Comment:
I don't like the use of "Props" in this name for reasons I noted elsewhere).

If this is just extra options for constructing schema, I think NodeSchemaOptions may be better.